Bottom: ebbbea4baf1eb2d90c2ec262de6265ed8b91b556
Top:    b6de5277eae4b45ebac00eb1c81693fdf6c9b8da
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-05-29 17:24:16 +0800

Refresh of vhost_net-split-out-datacopy

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index ed04efa..b3de4a0 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -494,7 +494,6 @@ static void handle_tx_copy(struct vhost_net *net)
 	int err;
 	size_t hdr_size;
 	struct socket *sock;
-	struct vhost_net_ubuf_ref *uninitialized_var(ubufs);
 	int sent_pkts = 0;
 
 	mutex_lock(&vq->mutex);
@@ -531,11 +530,17 @@ static void handle_tx_copy(struct vhost_net *net)
 			break;
 		}
 
+		/* Sanity check */
 		len = init_iov_iter(vq, &msg.msg_iter, hdr_size, out);
-		if (len < 0)
+		if (!len) {
+			vq_err(vq, "Unexpected header len for TX: "
+			"%zd expected %zd\n",
+			len, hdr_size);
 			break;
+		}
 
 		total_len += len;
+
 		if (total_len < VHOST_NET_WEIGHT &&
 		    !vhost_vq_avail_empty(&net->dev, vq))
 			msg.msg_flags |= MSG_MORE;
